cmd/gateway-main.go

```diff
@@ -37,18 +37,34 @@ func init() {
 	logger.RegisterUIError(fmtError)
 }
 
+const (
+	gatewayBucketFilterFlag = "bucket-filter"
+)
+
+var gatewayFlags = []cli.Flag{
+	cli.StringSliceFlag{
+		Name:   gatewayBucketFilterFlag,
+		Usage:  "whitelist patterns of buckets to expose which can include wildcard *",
+		EnvVar: "BUCKET_FILTER",
+	},
+}
+
+var globalGatewayContext = struct {
+	bucketFilter []string
+}{}
+
 var (
 	gatewayCmd = cli.Command{
 		Name:            "gateway",
 		Usage:           "start object storage gateway",
-		Flags:           append(ServerFlags, GlobalFlags...),
+		Flags:           append(gatewayFlags, append(ServerFlags, GlobalFlags...)...),
 		HideHelpCommand: true,
 	}
 )
 
 // RegisterGatewayCommand registers a new command for gateway.
 func RegisterGatewayCommand(cmd cli.Command) error {
-	cmd.Flags = append(append(cmd.Flags, append(cmd.Flags, ServerFlags...)...), GlobalFlags...)
```

Comment:
This appends `cmd.Flags` twice so that we can't have any flags at each gateway command.
(It's blocked by an error)